### ISSUE TITLE
⌨ Improper / missing labels on applicant dashboard

### ISSUE THREAD

**Issue by BrookeCameron** [GCTC-NTGC/gc-digital-talent#13394]
### ⌨ Accessibility and Expected Behaviour

Recruitment processes is labeled as a link but doesn't tell you what it does. - it should tell you that it is expandable. 

Headings under "your information" say the group is a link but don't activate when selected. the entire area should be a link and should activate when selected. 

### 📋 Steps to Reproduce

1. tab through items using VoiceOver 

### What operating systems are you encountering the problem on?

macOS

### What browsers are you encountering the problem on?

Firefox

### What assistive technologies are you encountering the problem on?

VoiceOver

### 📸 Screenshot

https://github.com/user-attachments/assets/719d5962-ccea-439f-83a0-bcf2520c9eaf
https://github.com/user-attachments/assets/ef5bc593-e4f5-46b4-b313-272189f59c3b

**Comment by petertgiles** [comment]
Oh accordions... sigh...  @esizer Is there anything new in this issue that isn't covered in other accordion issues on the board?

**Comment by esizer** [comment]
> Oh accordions... sigh... [@esizer](https://github.com/esizer) Is there anything new in this issue that isn't covered in other accordion issues on the board?

This might be a language thing or an invalid HTML thing. We have the proper markup to tell the user it controls an expandable area but the label is exceedingly long so maybe they never heard that? 🤔 

![Image](https://github.com/user-attachments/assets/7e148fc9-fbad-4f9b-a1a8-771930ade234)
